app/upgrades/v15/upgrades.go

```diff
@@ -64,33 +121,201 @@ func UpgradeSigningInfos(ctx sdk.Context, sk slashingkeeper.Keeper) {
 	}
 
 	ctx.Logger().Info("Finished migrating signing infos")
+	return nil
 }
 
-// UpgradeMinCommissionRate sets the minimum commission rate staking parameter to 5%
-// and updates the commission rate for all validators that have a commission rate less than 5%
-// adhere to prop 826 which sets the minimum commission rate to 5% for all validators
-// https://www.mintscan.io/cosmos/proposals/826
-func UpgradeMinCommissionRate(ctx sdk.Context, sk stakingkeeper.Keeper) {
-	ctx.Logger().Info("Migrating min commission rate...")
+// ClawbackVestingFunds transfers the vesting tokens from the given vesting account
+// to the community pool
+func ClawbackVestingFunds(ctx sdk.Context, address sdk.AccAddress, keepers *keepers.AppKeepers) error {
+	ctx.Logger().Info("Migrating vesting funds...")
 
-	params := sk.GetParams(ctx)
-	params.MinCommissionRate = sdk.NewDecWithPrec(5, 2)
-	err := sk.SetParams(ctx, params)
-	if err != nil {
-		panic(err)
+	ak := keepers.AccountKeeper
+	bk := keepers.BankKeeper
+	dk := keepers.DistrKeeper
+	sk := *keepers.StakingKeeper
+
+	// get target account
+	account := ak.GetAccount(ctx, address)
+
+	// verify that it's a vesting account type
+	vestAccount, ok := account.(*vesting.ContinuousVestingAccount)
+	if !ok {
+		ctx.Logger().Error(
+			"failed migrating vesting funds: %s: %s",
+			"provided account address isn't a vesting account: ",
+			address.String(),
+		)
+
+		return nil
 	}
 
-	for _, val := range sk.GetAllValidators(ctx) {
-		if val.Commission.CommissionRates.Rate.LT(sdk.NewDecWithPrec(5, 2)) {
-			// set the commission rate to 5%
-			val.Commission.CommissionRates.Rate = sdk.NewDecWithPrec(5, 2)
-			// set the max rate to 5% if it is less than 5%
-			if val.Commission.CommissionRates.MaxRate.LT(sdk.NewDecWithPrec(5, 2)) {
-				val.Commission.CommissionRates.MaxRate = sdk.NewDecWithPrec(5, 2)
+	// returns if no coins are vesting
+	_, vestingCoins := vestAccount.GetVestingCoins(ctx.BlockTime()).Find(sk.BondDenom(ctx))
```

Comment:
If GetVestingCoins returns nil, this will still fail as it will try to call Find on a nil pointer.

Reply:
Actually, `GetVestingCoins` never returns `nil` but an empty list of coins `sdk.Coins{}`.
Otherwise, this [test](https://github.com/cosmos/gaia/blob/6a22eee8e37979864c3991986e2ed16cc09fe05c/app/upgrades/v15/upgrades_test.go#L204) would have failed.

I refactored it to remove any ambiguity.